sdkproject/Assets/Mapbox/Unity/DataContainers/TileJsonData.cs

```diff
@@ -0,0 +1,191 @@
+﻿namespace Mapbox.Unity.Map
+{
+	using System;
+	using System.Collections.Generic;
+	using Mapbox.Platform.TilesetTileJSON;
+	using UnityEngine;
+
+	[Serializable]
+	public class TileJsonData
+	{
+		public readonly string commonLayersKey = "(layer found in more than one source)";
+		public readonly string optionalPropertiesString = "(may not appear across all locations)";
+		/// <summary>
+		/// This boolean is to check if tile JSON data has loaded after the data source has changed
+		/// </summary>
+		public bool tileJSONLoaded = false;
+		/// <summary>
+		/// Layer Display Names seen in the editor
+		/// </summary>
+		public List<string> LayerDisplayNames = new List<string>();
+
+
+		/// <summary>
+		/// Property Display Names seen in the editor
+		/// </summary>
+		public Dictionary<string, List<string>> PropertyDisplayNames = new Dictionary<string, List<string>>();
+
+		/// <summary>
+		/// The description of the property in a layer
+		/// </summary>
+		public Dictionary<string, Dictionary<string, string>> LayerPropertyDescriptionDictionary = new Dictionary<string, Dictionary<string, string>>();
+
+		/// <summary>
+		/// List of data sources (tileset ids) linked to a layer name
+		/// </summary>
+		public Dictionary<string, List<string>> LayerSourcesDictionary = new Dictionary<string, List<string>>();
+
+		/// <summary>
+		/// Dictionary containting the list of layers in a source
+		/// </summary>
+		public Dictionary<string, List<string>> SourceLayersDictionary = new Dictionary<string, List<string>>();
+
+		public void ClearData()
+		{
+			tileJSONLoaded = false;
+			LayerPropertyDescriptionDictionary.Clear();
+			LayerSourcesDictionary.Clear();
+			SourceLayersDictionary.Clear();
+			LayerDisplayNames.Clear();
+			PropertyDisplayNames.Clear();
+		}
+
+		public void ProcessTileJSONData(TileJSONResponse tjr)
+		{
+			List<string> layerPropertiesList = new List<string>();
+			List<string> sourceLayersList = new List<string>();
+
+			if (tjr == null || tjr.VectorLayers == null || tjr.VectorLayers.Length == 0)
+			{
+				return;
```

Comment:
Is there a loading or error state for this response?  It seems like editor scripts should know if a request is pending vs a request has returned null. 

In one case, things are working as expected and we're within tile timeout set in the setup window. In the other case, we should show an error.